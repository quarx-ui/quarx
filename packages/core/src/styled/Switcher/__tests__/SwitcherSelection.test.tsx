import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import userEvent from '@testing-library/user-event';
import { SwitcherSelection } from '@core';

describe('SwitcherSelection', () => {
    it('default snapshot', () => {
        const { asFragment } = render(<SwitcherSelection title="SwitcherSelection" />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('text should be in the document', () => {
        render(<SwitcherSelection title="SwitcherSelection" />);

        expect(screen.queryByText('SwitcherSelection')).toBeInTheDocument();
        const switcher = document.querySelector('input');
        expect(switcher).not.toHaveAttribute('disabled');
    });

    it('should be checked', () => {
        render(<SwitcherSelection checked title="SwitcherSelection" />);

        const switcher = document.querySelector('input');
        expect(switcher?.checked).toEqual(true);
    });

    it('should have attribute disabled', () => {
        const mockedOnClick = jest.fn();
        render(<SwitcherSelection disabled onClick={mockedOnClick} title="SwitcherSelection" />);
        const input = document.querySelector('input');
        const switcher = document.querySelector('.QxSwitcherSelection');

        if (!switcher || !input) { return; }

        expect(input).toHaveAttribute('disabled');
        userEvent.click(switcher);
        expect(mockedOnClick).not.toBeCalled();
    });

    it('should be hidden', () => {
        render(<SwitcherSelection hidden title="SwitcherSelection" />);
        expect(document.querySelector('.QxSwitcherSelection')).not.toBeInTheDocument();

        cleanup();

        render(<SwitcherSelection permissions={{ hidden: true }} title="SwitcherSelection" />);
        expect(document.querySelector('.QxSwitcherSelection')).not.toBeInTheDocument();
    });

    it('onClick should be called', () => {
        let checked = false;
        const mockedOnChange = jest.fn(() => {
            checked = !checked;
        });
        render(<SwitcherSelection onClick={mockedOnChange} title="SwitcherSelection" />);
        const switcher = document.querySelector('.QxSwitcherSelection');

        if (!switcher) { return; }

        userEvent.click(switcher);
        expect(mockedOnChange).toBeCalledTimes(1);
        expect(checked).toEqual(true);
    });
});
