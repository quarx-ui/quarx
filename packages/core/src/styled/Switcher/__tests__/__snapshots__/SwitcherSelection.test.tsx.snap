// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitcherSelection default snapshot 1`] = `
<DocumentFragment>
  <div
    class="QxSelection QxSelection_size_medium QxSelection_styleHover_default QxSelection_borderRadius_smooth QxSwitcherSelection css-1e56tu3-QxSelection-root"
  >
    <div
      class="QxSelection-leftAdornment css-1xv7td9-QxSelection-leftAdornment"
    >
      <label
        class="QxSwitcher QxSwitcher_size_medium QxSwitcher_color_brand QxSwitcher_disableFocus QxSwitcher_position_left css-1dfsk5i-QxSwitcher-root"
      >
        <input
          class="QxSwitcher-input css-1l6pf64-QxSwitcher-input"
          type="checkbox"
          value=""
        />
        <div
          class="QxSwitcher-toggleContainer css-1pxim4o-QxSwitcher-toggleContainer"
        >
          <div
            class="QxSwitcher-toggle css-1c28ujy-QxSwitcher-toggle"
          />
        </div>
      </label>
    </div>
    <div
      class="QxSelection-text css-4p6ohz-QxSelection-text"
    >
      <div
        class="QxSelection-title css-11y2um3-QxSelection-title"
      >
        SwitcherSelection
      </div>
    </div>
  </div>
</DocumentFragment>
`;
